src/js/background.js

```diff
@@ -0,0 +1,214 @@
+  });
+};
+
+const waitForElement = async (
+  querySelector,
+  tabId,
+  timeLeft = 5000,
+  retryDelay = 100
+) => {
+  const elementFoundScript = `document.querySelector("${querySelector}") !== null;`;
+
+  let isElement = null;
+
+  while (!isElement && timeLeft > 0) {
+    await new Promise((r) => setTimeout(r, retryDelay));
+    timeLeft -= retryDelay;
+    [isElement] = await browser.tabs.executeScript(tabId, {
+      code: elementFoundScript,
+    });
+  }
+
+  if (!isElement)
+    sendLogMessage(`Cannot find \`${querySelector}\` element!`, tabId);
+
+  return isElement;
+};
+
+const waitAndClick = async (querySelector, tabId) => {
+  if (await waitForElement(querySelector, tabId))
+    await clickElement(querySelector, tabId);
+};
+
+const lichessAnalyse = async (tabId, pgn) => {
+  if (await waitForElement("[name='analyse']", tabId)) {
+    const [loggedIn] = await browser.tabs.executeScript(tabId, {
+      code: `!document.querySelector("[name='analyse']").disabled;`,
+    });
+    if (loggedIn) await waitAndClick("[name='analyse']", tabId);
+
+    if (await waitForElement("[name='pgn']", tabId)) {
+      await browser.tabs.executeScript(tabId, {
+        code: `document.querySelector("[name='pgn']").value = \`${pgn}\`;`,
+      });
+      await waitAndClick(".submit", tabId);
+
+      /*
+       * Bugfix where status was loading and firefox throwed
+       * an error about invalid host permissions
+       */
+      let status = "loading";
+      while (status === "loading") {
+        let getting = await browser.tabs.get(tabId);
+        status = getting.status;
+      }
+
+      if (await waitForElement("#analyse-toggle-ceval", tabId)) {
+        const [localEval] = await browser.tabs.executeScript(tabId, {
+          code: `document.querySelector("#analyse-toggle-ceval").checked;`,
+        });
+        if (!localEval)
+          await waitAndClick("[for='analyse-toggle-ceval']", tabId);
+      }
+    }
+  }
+};
+
+/*
+ * Structure:
+ * {
+ *   tabId0: bool,
+ *   tabId1: bool
+ * }
+ */
+let analysingState = {};
+
+const analyseGame = async (tab) => {
+  /* After clicking on pageAction twice, second call won't be executed */
+  if (analysingState[tab.id]) return;
+  analysingState[tab.id] = true;
+
+  try {
+    await setLoadingState(true, tab.id);
+
+    const [playerName] = await browser.tabs.executeScript(tab.id, {
+      code: `document.querySelector('[data-test-element="user-tagline-username"]').textContent;`,
+    });
+
+    const gameId = tab.url.split("?")[0];
+
+    let pgn = await getPGN(playerName, gameId, tab.id);
+
+    await setLoadingState(false, tab.id);
+
+    if (!pgn) {
+      sendLogMessage(
+        `Game with id ${gameId} not found! Performing manual fetching.`,
+        tab.id
+      );
+      pgn = await getPGNManual(tab.id);
+    }
+
+    if (!pgn) {
+      sendLogMessage(`Game not found!`, tab.id);
+      analysingState[tab.id] = false;
+    }
```

Comment:
Are you sure we can continue ot creating a lichess tab even if `pgn` was not found?